assets/src/edit-story/components/IconSwitch/index.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * External dependencies
+ */
+import styled from 'styled-components';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import LockClosedIcon from './lock_closed.svg';
+import LockOpenIcon from './lock_open.svg';
+
+const StyledButton = styled.button`
+  border: none;
+  background: transparent;
+  display: block;
+	padding: 0;
+	width: ${ ( { width } ) => width }px;
+	height: ${ ( { height } ) => height }px;
+  min-width: initial;
+  cursor: pointer;
+	color: ${ ( { theme } ) => theme.colors.fg.v3 };
+	&:focus, &:active, &:hover {
+    opacity: 1;
+    outline: none;
+	}
+	svg {
+		width: ${ ( { width } ) => width }px;
+		height: ${ ( { height } ) => height }px;
+  }
+  ${ ( { disabled } ) => disabled && `
+		pointer-events: none;
+		opacity: .3;
+	` }
+`;
+
+const IconSwitch = ( { locked, ...props } ) => (
+	<StyledButton { ...props }>
+		{ locked ? <LockClosedIcon /> : <LockOpenIcon /> }
```

Comment:
The `IconSwitch` should work for any two icons - not just this lock icon. Make it more abstract, but do expose a `LockSwitch` component concretizing it to the lock icon. Also, `checked` is the common variable used to indicate "checked"-ness.

Reply:
I tried to make a more general IconSwitch component with LockSwitch component that uses IconSwitch. IconSwitch currently doesn't have default icons for status which will cause some errors if we don't define icons when using IconSwitch.  Do you think we need default icons?